spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskControllerTests.java

```diff
@@ -222,4 +224,31 @@ public void testLaunch() throws Exception {
 		assertEquals("1", mavenResource.getVersion());
 		assertEquals("myTask", request.getDefinition().getProperties().get("spring.cloud.task.name"));
 	}
+
+	@Test
+	public void testLaunchWithParams() throws Exception {
+		repository.save(new TaskDefinition("myTask2", "foo2"));
+		this.registry.register("task.foo2", new URI("maven://org.springframework.cloud:foo2:1"));
+
+		mockMvc.perform(
+				post("/tasks/deployments/{name}", "myTask2")
+				.param("params", "--foobar=jee")
```

Comment:
When running a batch app via boot, all the args are passed in as job parameters.